AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -116,6 +118,32 @@ - (void)dealloc
 
 #pragma mark - Public methods -- must lock
 
+/// Setter for public image property. It has the side effect to set an internal _imageWasSetExternally that prevents setting an image internally. Setting an image internally should happen with the _setImage: method
+- (void)setImage:(UIImage *)image
+{
+  __instanceLock__.lock();
+  
+#ifdef DEBUG
+  if (_URL != nil) {
+    NSLog(@"Setting the image directly on an %@ and setting and setting an URL is not supported. If you want to use a placeholder image please use defaultImage .", NSStringFromClass([self class]));
```

Comment:
Is this necessary? If not, I think we should support this condition. Think for example of using a network image node in a transition, and so you might want to override its image with some other one temporarily, to achieve a visual effect.